dpnp/dpnp_iface_histograms.py

```diff
@@ -306,6 +307,66 @@ def digitize(x, bins, right=False):
     return bins.size - dpnp.searchsorted(bins[::-1], x, side=side)
 
 
+def _find_supported_dtype(dt, supported):
+    if dt in supported:
+        return dt
+
+    for t in supported:
+        if dpnp.can_cast(dt, t):
+            return t
+
+    return None
+
+
+def _result_type(dtype1, dtype2, has_fp64):
+    rt = dpnp.result_type(dtype1, dtype2)
+    if rt == dpnp.float64 and not has_fp64:
```

Comment:
There is `map_dtype_to_device` function available that can be used for such cases. Using it has this benefit that the coverage report, will not show any missing line as it does currently.

`dpnp/dpnp_iface_histograms.py, coverage:   93%  missing lines: 213, 318, 324, 327, 336->339, 358, 359->362, 364-365, 473`

Reply:
Thanks! I've modified `_result_type` to use `map_dtype_to_device`